dsgp/dsgp-005.md

```diff
@@ -0,0 +1,174 @@
+# DSGP 5 -- First statements
+
+| Author        | Author email         | Status   |
+|---------------|----------------------|----------|
+| Diego Ramirez | dr01191115@gmail.com | Pending  |
+
+## Overview
+
+In the 1.x series of [DiddiParser 2][1], having statements was just impossible, unless
+you figured out a way using functions.
+
+But after [DSGP 4][2] acceptance, there's a chance for a new feature of the DiddiScript syntax:
+**statements**. However, that DSGP didn't define any statement, it only defined the concept.
+
+This DSGP is the DSGP 4 partner, and defines the first bunch of statements, which are statements
+accepted by several languages.
+
+## The `if/elseif/else` statements
+
+The `if/elseif/else` can be easily recognized, because multiple languages accept the `if/else`
+structure, and many of them have the `elseif` as the something-in-between (in other languages,
+`elseif` is also known as `elif` or `else if`).
+
+The syntax, based in the DSGP 4 syntax, goes as follows:
+
+```
+if (boolean) {
+    !# steps
+}
+elseif (boolean) {
+    !# steps
+}
+else {
+    !# steps
+}
+```
+
+Since it is a set of dependent statements, they should combine to achieve certain goals.
+The following marked combinations are allowed. Unmarked combinations are considered syntax errors:
+
+- [x] `if`
+- [x] `if` / `else`
+- [x] `if` / multiple `elseif`s
+- [x] `if` / multiple `elseif`s / `else`
+- [ ] multiple `elseif`s
+- [ ] `else`
+- [ ] multiple `elseif`s / else
+- [ ] `if` / statements different than `if/elseif/else` / `else`
+- [ ] statements different than `if/elseif/else` / `else`
+- [ ] statements different than `if/elseif/else` / `elseif`
+
+First of all, there should be one `if` (no orphan `elseif` or `else` blocks are allowed).
+There shuld be one `else` block (or none). An infinite number of `elseif` blocks are allowed.
+
+## The `while` statement
+
+The `while` statement is another well-accepted loop. It repeats the nested steps while the
+condition is true. The syntax is the following:
+
+```
+while (boolean) {
+    !# steps
+}
+```
+
+The way conditions are evaluated is:
+
+- Check the condition
+  - If true, run the nested steps.
+  - If false, break the loop.
+
+The flow can be affected by `break` and `continue` instructions (see below). Infinite loops can
+happen (if the condition doesn't change and keeps true).
+
+## The `repeat` statement
+
+The `repeat` statement will repeat steps.
+
+Unlike `while`, `repeat` won't analyze a condition. Instead, it will repeat the steps for a strict
+number of steps (integer or floating numbers are expected). For example, if `10` is passed, the
+steps will be repeated 10 times.
+
+This is the syntax:
+
+```
+repeat (integer_or_floating) {
+    !# steps
+}
+```
+
+The flow can be modified by a `break` or `continue` statement (see below). In the case of `continue`,
+the counter will be increased, as it is a new iteration.
+
+## An addition independent to the DSGP 4 spec: `break` and `continue`
+
```

Comment:
Ok, this is independent to DSGP 4, but it is important to go ahead with other statements :)